frontends/p4/typeMap.cpp

```diff
@@ -160,15 +160,23 @@ bool TypeMap::equivalent(const IR::Type *left, const IR::Type *right, bool stric
         return lv->getVarName() == rv->getVarName() && lv->getDeclId() == rv->getDeclId();
     }
     if (auto ls = left->to<IR::Type_Stack>()) {
+        const auto isLegal = [](const IR::Type_Stack *stack) {
```

Comment:
I'm surprised the checks are in this place, this does not seem the right place to put them to me. I don't even see why this placement guarantees the check will run during compilation.

Maybe this could be placed into type checking (although that one is very long already :-/), or into some other validation pass (it should run after constant folding though).